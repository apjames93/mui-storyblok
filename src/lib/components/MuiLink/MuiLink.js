import React from 'react';
import PropTypes from 'prop-types';
import Link from '@material-ui/core/Link';
import Storyblok from '../../utils/Storyblok';

/**
 * MuiLink is used in storyblok redirect to a href
 */

export const MuiLink = ({
  text,
  color,
  href,
  underline,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  const onClick = () => {
    window.location.replace(href);
  };

  return (
    // eslint-disable-next-line jsx-a11y/anchor-is-valid
    <Link
      className={styles.root}
      color={color}
      underline={underline}
      href="#"
      onClick={onClick}
    >
      {text}
    </Link>
  );
};

export default MuiLink;

MuiLink.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop:
   'initial' | 'inherit' | 'primary' | 'secondary' | 'textPrimary' | 'textSecondary' | 'error'
   * The color of the component. It supports those theme colors that make sense for this component.
   * */
  color: PropTypes.string,
  /*
  * mui prop:'none' | 'hover' | 'always'
  * underline text
  */
  underline: PropTypes.string,
  /* url for redirect */
  href: PropTypes.string.isRequired,
  /* text to click on */
  text: PropTypes.string.isRequired,
};

MuiLink.defaultProps = {
  color: 'primary',
  underline: 'hover',
  rootClass: [],
};
