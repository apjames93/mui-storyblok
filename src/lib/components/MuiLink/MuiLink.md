MuiLink Examples:

```js
const underline = 'hover';
const href = 'https://github.com/apjames93/mui-storyblok';
const text = 'MuiStoryblok Github';

<MuiLink
  underline={underline}
  href={href}
  text={text}
/>

```