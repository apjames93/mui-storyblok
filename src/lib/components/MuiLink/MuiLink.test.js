import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiLink from './MuiLink';

function setup() {
  const props = {
    href: 'href.com',
    text: 'text',
  };
  const comp = shallow(<MuiLink {...props} />);
  return { comp, props };
}

describe('<MuiLink />', () => {
  it('renders MuiLink', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('renders MuiLink', () => {
    const { replace } = window.location;
    window.location.replace = jest.fn();
    const { comp } = setup();
    comp.find('WithStyles(ForwardRef(Link))').first().simulate('click');
    expect(window.location.replace).toBeCalled();
    window.location.replace = replace;
  });


  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiLink {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
