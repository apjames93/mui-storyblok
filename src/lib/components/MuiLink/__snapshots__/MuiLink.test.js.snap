// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiLink /> snapshot 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-root-4 MuiTypography-colorPrimary"
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  text
</a>
`;
